src/main/java/rueppellii/backend2/tribes/kingdom/service/KingdomService.java

```diff
@@ -51,8 +64,16 @@ public Kingdom findByPrincipal(Principal principal) throws KingdomNotFoundExcept
         return findByUsername(loggedInUser);
     }
 
-    public Kingdom createNewKingdomAndSetNameIfNotExists(ApplicationUserDTO applicationUserDTO) {
+    public Kingdom createNewKingdomAndSetNameIfNotExists(ApplicationUserDTO applicationUserDTO) throws IOException, CountryCodeNotValidException, LocationIsTakenException {
```

Comment:
refactor this method, if the method's name contains "and", it doesn't sound great :) Talk about it